### PR TITLE
Add infrastructure for search methods

### PR DESCRIPTION
**Pull request by pakrym-stripe**
https://jira.corp.stripe.com/browse/DEVSDK-526

r? @dcr-stripe 

Adds infrastructure required to generate Search methods.

Adds mock tests we can add stripe-mock coverage when it's updated.

### CODE REVIEW

File: src/Stripe.net/Entities/StripeSearchResult.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        /// The URL for accessing this list.
+        /// </summary>
+        [JsonProperty("url")]
+        public string Url { get; set; }
+
+        /// <summary>
+        /// The URL for accessing the next page in search results.
+        /// </summary>
+        [JsonProperty("page")]
+        public string Page { get; set; }
```
Comment (dcr-stripe) [inline src/Stripe.net/Entities/StripeSearchResult.cs:40]: Results will continue to return `next_page`. Only the request uses `page`.

File: src/Stripe.net/Services/_base/SearchOptions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    public class SearchOptions : BaseOptions
+    {
+        /// <summary>
+        /// A limit on the number of objects to be returned, between 1 and 100.
+        /// </summary>
+        [JsonProperty("limit")]
+        public long? Limit { get; set; }
+
+        [JsonProperty("next_page")]
+        public string NextPage { get; set; }
```
Comment (dcr-stripe) [inline src/Stripe.net/Services/_base/SearchOptions.cs:14]: This should be `page`.

**Comment by pakrym-stripe** [comment]
>Have you done an end to end test with this against an actual end point? 

I did. I'm looking into why I didn't catch this.

**Comment by pakrym-stripe** [comment]
re-r? @dcr-stripe

**Pull request merged**